pxc/jenkins/qa_pxc_cross-pipeline.groovy

```diff
@@ -0,0 +1,180 @@
+        } //End stage Debian Bullseye
+        stage('Debian Buster') {
+          agent {
+            label "min-buster-x64"
+          }
+          steps {
+            withCredentials([usernamePassword(credentialsId: 'JenkinsAPI', passwordVariable: 'JENKINS_API_PWD', usernameVariable: 'JENKINS_API_USER')]) {
+              run_test()
+            }
+          } //End steps
+        } //End stage Debian Buster
+        stage('Centos7') {
+          agent {
+            label "min-centos-7-x64"
+          }
+          steps {
+            withCredentials([usernamePassword(credentialsId: 'JenkinsAPI', passwordVariable: 'JENKINS_API_PWD', usernameVariable: 'JENKINS_API_USER')]) {
+              run_test()
+            }
+          } //End steps
+        } //End stage CentOS7
+        stage('Oracle Linux 8') {
+          agent {
+            label "min-ol-8-x64"
+          }
+          steps {
+            withCredentials([usernamePassword(credentialsId: 'JenkinsAPI', passwordVariable: 'JENKINS_API_PWD', usernameVariable: 'JENKINS_API_USER')]) {
+              run_test()
+            }
+          } //End steps
+        } //End stage Oracle Linux 8
+       } //End parallel
+    } //End stage Run tests
+  } //End stages
+} //End pipeline
+
+void run_test() {
+  sh 'echo Downloading LOWER_PXC tarball: \$(date -u)'
+               sh '''
+                    echo "Installing dependencies..."
+                    if [ -f /usr/bin/yum ]; then 
+                    sudo yum -y update
+                    sudo yum install -y git wget tar socat
+                    else
+                    sudo apt install -y git wget ansible socat curl numactl
+                    sudo apt update
+                    curl -O https://repo.percona.com/apt/percona-release_latest.generic_all.deb
+                    sudo apt install -y gnupg2 lsb-release ./percona-release_latest.generic_all.deb
+                    sudo apt update
+                    fi
+                    ROOT_FS=$(pwd)
+                    sudo killall -9 mysqld || true
+                    # Fetch the latest LOWER_PXC binaries
+                    cd $ROOT_FS/
+                    rm -rf $ROOT_FS/pxc_5.7_tar || true
+                    LOWER_PXC_TAR=lower-pxc-latest.tar.gz
+                    wget -qcO - https://downloads.percona.com/downloads/TESTING/pxc-${LOWER_PXC_VERSION}/Percona-XtraDB-Cluster-${PXC57_PKG_VERSION}.Linux.x86_64.glibc2.17.tar.gz > ${LOWER_PXC_TAR}
+                    tar -xzf ${LOWER_PXC_TAR}
+                    rm *.tar.gz
+                    mv Percona-XtraDB-* pxc_5.7_tar
+                '''
+  sh 'echo Downloading Upper_PXC tarball: \$(date -u)'
+               sh '''
+                    ROOT_FS=$(pwd)
+                    sudo killall -9 mysqld || true
+                    # Fetch the latest Upper_PXC binaries
+                    cd $ROOT_FS
+                    rm -rf $ROOT_FS/pxc_8.0_tar || true
+                    UPPER_PXC_TAR=upper-pxc-latest.tar.gz
```

Comment:
Maybe rename the variable to something better like
PXC_UPPER_VERSION_TAR
PXC_LOWER_VERSION_TAR

Reply:
Fixed !